src/borg/archiver.py

```diff
@@ -521,7 +528,8 @@ def _process(self, archive, cache, matcher, exclude_caches, exclude_if_present,
                                               read_special=read_special, dry_run=dry_run)
                         return
                 if not dry_run:
-                    status = archive.process_dir(path, st)
+                    if not dir_nomatch:
+                        status = archive.process_dir(path, st)
```

Comment:
hmm, didn't you say you wanted to recurse into the excluded dirs to search for presence of maybe included dirs below them?